src/data_api/reporting.se

```diff
@@ -27,186 +27,161 @@ extern branches: [addCurrency:[int256,int256,int256,int256]:int256, addMarketToB
 
 data controller
 
 def init():
     self.controller = 0x0
     REPCONTRACT = self.controller.lookup('repContract')
-    self.Reporting[1010101].repIDtoIndex[1010101] = 0
     redistributionRep = REPCONTRACT.balanceOf(0x0000000000000000000000000000000000000000)
-    self.Reporting[1010101].reputation[0].repValue = redistributionRep
-    self.Reporting[1010101].reputation[0].reporterID = 1010101
-    self.Reporting[1010101].dormantRep[0].reporterID = 1010101
-    self.Reporting[1010101].numberReporters = 2
-    self.Reporting[1010101].repIDtoIndex[msg.sender] = 1
-    self.Reporting[1010101].reputation[1].repValue = 0
-    self.Reporting[1010101].reputation[1].reporterID = msg.sender
-    self.Reporting[1010101].dormantRep[1].reporterID = msg.sender
-    self.Reporting[1010101].repIDtoIndex[msg.sender] = 1
-    self.Reporting[1010101].activeRep += redistributionRep
-    self.Reporting[1010101].totalRep += redistributionRep
-
-def getReportedOnNonFinalRoundTwoEvent(branch, reporter):
-    return(self.Reporting[branch].reportedOnNonFinalRoundTwoEvent[reporter])
-
-def setReportedOnNonFinalRoundTwoEvent(branch, event, reporter):
+    self.Reporting[1010101].activeRep[0x0000000000000000000000000000000000000000] = 0
+    self.Reporting[1010101].dormantRep[0x0000000000000000000000000000000000000000] = redistributionRep
+    self.Reporting[1010101].totalActiveRep = 0
+    self.Reporting[1010101].totalDormantRep = redistributionRep
+    # TODO: figure out why address 0 counts as a reporter, feels like this should be `0`
+    self.Reporting[1010101].numberReporters = 1
+    self.Reporting[1010101].reporterLocks[0x0000000000000000000000000000000000000000].eventCount = 0
+
+def getNumberOfEventsLockingReporter(branch, reporterAddress):
+    return(self.Reporting[branch].reporterLocks[reporterAddress].eventCount)
+
+def lockReporter(branch, reporterAddress, event):
     self.controller.checkWhitelist(msg.sender)
-    self.Reporting[branch].reportedOnNonFinalRoundTwoEvent[reporter] = event
+    self.Reporting[branch].reporterLocks[reporterAddress].events[event] = 1
+    self.Reporting[branch].reporterLocks[reporterAddress].eventCount += 1
     return(1)
 
-# @return fxp
-def getActiveRep(branch):
-    return(self.Reporting[branch].activeRep)
+def unlockReporter(branch, reporterAddress, event):
+    self.controller.checkWhitelist(msg.sender)
+    if (self.Reporting[branch].reporterLocks[reporterAddress].events[event] == 0):
+        throw()
+    if (self.Reporting[branch].reporterLocks[reporterAddress].eventCount <= 0):
+        throw() # if we are in this state things are _really_ broken
+    self.Reporting[branch].reporterLocks[reporterAddress].events[event] = 0
+    self.Reporting[branch].reporterLocks[reporterAddress].eventCount -= 1
+    return(1)
 
-def adjustActiveRep(branch, fxpAmount):
+def activateRep(branch, attorep):
```

Comment:
Or perhaps the argument is that activating/deactivating rep isn't an action that a user would every want to take, instead it is intended to be an automated process (in which case I'm probably not fully grokking active/dormant rep).  I thought a user would generally have REP dormant when they wanted to trade it or were unable to report (e.g., vacation) and then when they wanted to start receiving fees they would activate.